src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1411 @@
+                <em>Failed to retrieve External Source metadata.</em>
+              {:else}
+                <em>Source has no metadata.</em>
+              {/if}
+            {:catch error}
+              <em>error loading metadata...try refreshing the page.</em>
+              {catchError(error)}
+            {/await}
+          </Collapse>
+          <div style="padding-bottom:20px">
+            <Collapse
+              className="used-in-plans-collapse"
+              defaultExpanded={false}
+              title="Used in plans"
+              tooltipContent="View plans this source is used in"
+            >
+              {#if selectedSourceLinkedDerivationGroupsPlans.length > 0}
+                {#each selectedSourceLinkedDerivationGroupsPlans as linkedPlanDerivationGroup}
+                  <i>
+                    <a href="{base}/plans/{linkedPlanDerivationGroup.plan_id}">
+                      {$plans.find(plan => {
+                        return linkedPlanDerivationGroup.plan_id === plan.id;
+                      })?.name}
+                    </a></i
+                  >
+                {/each}
+              {:else}
+                <i>Not used in any plans</i>
+              {/if}
+            </Collapse>
+          </div>
+          <button
+            class="st-button danger w-100"
+            style="margin-bottom:auto;"
+            use:permissionHandler={{
+              hasPermission: hasDeletePermission,
+              permissionError: deletePermissionError,
+            }}
+            on:click|stopPropagation={async () => onDeleteExternalSource(selectedSource)}
+          >
+            Delete external source
+          </button>
+        </div>
+      {:else}
+        <form
+          on:submit|preventDefault={onFormSubmit}
+          on:reset={() => {
+            parsed = undefined;
+            $createExternalSourceError = null;
+            $createExternalSourceTypeError = null;
+            $createDerivationGroupError = null;
+            $parsingError = null;
+            isDerivationGroupFieldDisabled = true;
+          }}
+        >
+          <AlertError class="m-2" error={$createExternalSourceError} />
+          <AlertError class="m-2" error={$createExternalSourceTypeError} />
+          <AlertError class="m-2" error={$createDerivationGroupError} />
+          <AlertError class="m-2" error={$parsingError} />
+          <div id="file-upload-field">
+            <fieldset style="width:100%">
+              <label for="file">Source File</label>
+              <input
+                class="w-100"
+                name="file"
+                required
+                type="file"
+                bind:files
+                use:permissionHandler={{
+                  hasPermission: hasCreatePermission,
+                  permissionError: createPermissionError,
+                }}
+              />
+            </fieldset>
+
+            <fieldset id="file-upload-fieldset">
+              <button
```

Comment:
Consider moving these two buttons inside of the source file fieldset, grouping the input and label into a div and then the button group into a div.